feat(bivariate-matrix): 11607 - show ui response tooltip

propakov opened the pull request:
- fade animation is added to features/tooltip constantly

https://kontur.fibery.io/Tasks/Task/UI-response-for-selecting-layers-11607

<img width="615" alt="Screenshot 2022-09-05 at 17 21 53" src="https://user-images.githubusercontent.com/20676525/188470525-bd849f12-3dbf-4f8c-b48c-fd20a325f1fa.png">

propakov committed:
feat(bivariate-matrix): 11607 - show ui response tooltip when click on header/cell of matrix
- fade animation is added to features/tooltip constantly

IlyaIzr commented on src/features/bivariate_manager/components/ConnectedBivariateMatrix/ConnectedBivariateMatrix.tsx:
```diff
@@ -37,151 +39,161 @@ const mapHeaderCell = (
+    };
+  }, [stats, xGroups, yGroups]);
+
+  const debouncedShowTooltip = useCallback(
+    debounce((position) => {
+      resetTooltip();
+      setTooltip({
+        popup: <ProgressTooltip close={resetTooltip} />,
+        position,
+        initiatorId: TOOLTIP_ID,
+        hoverBehavior: true,
+      });
+    }, 400),
+    [resetTooltip, setTooltip],
+  );
```
Can you explain why do you use debounce?

propakov replied:
same reason of using in inputs, here we can select cells/heading fast, we don't need this tooltip on every select, just in the end.

IlyaIzr commented on src/features/bivariate_manager/components/ProgressTooltip/ProgressTooltip.tsx:
```diff
@@ -0,0 +1,52 @@
+    />
+  </svg>
+);
+
```
I believe you should ask designer to add icon into icons pack

IlyaIzr commented on src/features/tooltip/index.tsx:
```diff
@@ -1,8 +1,22 @@
+import {
+  CSSTransitionWrapper,
+  fadeClassNames,
+} from '~features/bivariate_color_manager/components/CssTransitionWrapper/CssTransitionWrapper';
```
❗ import into feature from another feature

propakov replied:
oh, really  :/ 
nice catch! 👁️ 💎

propakov committed:
feat(bivariate-matrix): 11607 - CR fixes
- CSSTransitionWrapper moved to components
- LoadingSpinner16 icon is used from ui-kit
